### PR TITLE
Add CodeChecks support

### PR DESCRIPTION
**Pull request by krzkaczor**
Adds a codechecks integration which allows plugging into Github PR flow. Related: #124 

[Sample PR](https://github.com/krzkaczor/cc-prod-playground/pull/5)

image

image

[About CodeChecks](https://github.com/codechecks/docs/blob/master/getting-started.md)

Configuration, for now, only supports simple file paths. `codechecks.yml`:
```yml
checks:
  - name: size-limit/codecheck
    options:
      files:
        - path: "build/**/*.js"

```

### CODE REVIEW

**Comment by ai** [comment]
Few stupid questions since I am new in CodeCheck:

1. Can we have Size Limit name and logo in the check?
2. Do we have any Size Limit page in CodeCheck page? Maybe settings page with the name?
3. Does current implementation use project limits?

(I will read more about CodeCheck today)

**Comment by krzkaczor** [comment]
Thanks for getting back to me so quickly!

1. Particular codechecks can change everything apart from logo — they all share one GitHub app. [Docs](https://github.com/codechecks/docs/blob/master/client-api.md#reportreport-codechecksreport-promise)
2. CodeChecks as a whole platform don't know anything about size-limit particular. Plugins should be sefl contained and any settings should be part of `codechecks.yml ` as shown [here](https://github.com/krzkaczor/cc-prod-playground/blob/master/codechecks.yml)
3. I am not sure if I understand but do you mean private users on GitHub? No, Codechecks is a GitHub app so it's doesn't take a "spot" in a Github Org / private repo. Btw. support for private repos is in closed beta so if you want to get it enabled just let me know :) 

To sum up: users of codechecks platform can utilize many different codechecks and it requires installing only one GitHub app. All configuration is stored inside repo so it's properly versioned, copy to new project etc.

Let me know if you have *any* questions. CodeChecks is really fresh project and I just started reaching out to the broader community for feedback.

**Comment by krzkaczor** [comment]
@ai btw. I think tests are broken on `master` and they don't pass on this branch too :(

**Comment by ai** [comment]
Thanks for clarifying.

The third question was about bundle size limit from `.size-limit.json`config. We need to fail build if this limit was exceeded.

BTW, I planned big refactoring for today. I will split config loading, UI and move running time calculation in separated package. Is it OK for you to update PR tomorrow. I think this refactoring will be useful for this feature as well.

**Comment by krzkaczor** [comment]
Ahh I see, so 3 is totally possible ;) Refactoring would be nice, there are lots of reusable functions in `cli-body.js` but I needed to extract them as requiring this file kicks of whole CLI. It would be cool if config loading functions would be extracted so I can easily use them in the codecheck.

**Comment by krzkaczor** [comment]
@ai could you let me know whats the status of a before-mentioned refactoring?

**Comment by krzkaczor** [comment]
@ai okay this makes sense. Maybe with these smaller packages in makes more sense to improve already existing codecheck: https://github.com/codechecks/build-size-watcher

**Comment by ai** [comment]
Yeap. It is another thing that I am worry about. They already have some solution. It may not covered few thing (works only with project with bundler).

**Comment by ai** [comment]
I finally started the refactoring and I like the new architecture.

I have the plan to move CodeChecks support to `@size-limit/code-checks` modules in a new modular architecture (to reduce dependencies).

**Comment by krzkaczor** [comment]
@ai sounds awesome! Let me know if I can help you with anything.

**Comment by ai** [comment]
2.0 was released

**Comment by krzkaczor** [comment]
@ai with my PR I wanted to somehow pave a way for codechecks integration. I am busy myself with maintaining the core of codechecks project. 

If integration with GitHub is something that you're interested please just continue work from where I left off. In case of any troubles, just ping me. 

Best!
